Automattic/wp-calypso
# Hosting onboarding: add site options step to `/setup/new-hosted-site`

Closes https://github.com/Automattic/dotcom-forge/issues/2360.

## Proposed Changes

This PR adds the data center picker, along with a site title suggestion, to the new `/hosting` onboarding flow. See the issue for details.

![image](https://github.com/Automattic/wp-calypso/assets/26530524/47b7f884-71a1-4103-a7d2-20d3dc13167f)

## Testing Instructions

1. Patch your sandbox with this diff: D110494-code;
2. Checkout this branch;
3. Browse `/setup/new-hosted-site`;
4. Verify that the site title gets auto-populated with a suggestion;
5. Verify that the recommended data picker is selected, but you can pick a custom one;
6. Buy the plan;
7. Verify that the site title is the same as the one suggested in the first step and that the domain slug is related to the suggestion;
8. Verify that the picked data center was communicated correctly*.

*You can use this snippet to log the picked data center right after checkout. Add this to `0-sandbox` **before pressing the Pay button**:

```php
function sandbox_activate_product( $blog_id, $user_id, $product_id, $txn_id, $verb, $meta, $volume, $extra ) {
	l( $extra );
}

add_action( 'activate_product', 'sandbox_activate_product', 9, 8 );
```

And you should see `{ extra: { geo_affinity: 'chosen_data_center' } }` on your sandbox logs.
